internal/client/githubclient/githubclient.go

```diff
@@ -111,6 +122,139 @@ func (gc *githubClient) GetLatestRelease(ctx context.Context, owner string, repo
 	return &release, nil
 }
 
+// GetWorkflow retrieves the workflow for a specified workflow name from a given GitHub repository.
+// If the workflow name is not provided, it fetches all workflows for the repository.
+// It returns an error if the workflow name is provided but not found in the repository.
+func (gc *githubClient) GetWorkflow(ctx context.Context, owner, repo, githubWorkflowFileName string) ([]*client.Workflow, error) {
+	if githubWorkflowFileName != "" {
+		workflow, _, err := gc.ghClient.Actions.GetWorkflowByFileName(ctx, owner, repo, githubWorkflowFileName)
+		if err != nil {
+			return nil, fmt.Errorf("unable to get workflow by file name: %w", err)
+		}
+
+		return []*client.Workflow{
+			{
+				Name: *workflow.Name,
+				Id:   *workflow.ID,
+			},
+		}, nil
+	}
+
+	workflows, _, err := gc.ghClient.Actions.ListWorkflows(ctx, owner, repo, nil)
+	if err != nil {
+		return nil, fmt.Errorf("unable to list workflows: %w", err)
+	}
+
+	var res []*client.Workflow
+
+	for _, workflow := range workflows.Workflows {
+		res = append(res, &client.Workflow{
+			Name: *workflow.Name,
+			Id:   *workflow.ID,
+		})
+	}
+
+	return res, nil
+}
+
+// GetLatestWorkflowRun retrieves all the workflow runs associated with a specified workflow ID from a given GitHub repository.
+// It returns an error if the workflow runs cannot be fetched.
+func (gc *githubClient) GetLatestWorkflowRun(ctx context.Context, owner, repo string, workflowId int64) (*client.WorkflowRun, error) {
+	runs, _, err := gc.ghClient.Actions.ListWorkflowRunsByID(ctx, owner, repo, workflowId, nil)
+	if err != nil {
+		return nil, fmt.Errorf("unable to list workflow runs: %w", err)
+	}
+
+	if len(runs.WorkflowRuns) == 0 {
+		return nil, nil
+	}
+
+	// runs.WorkflowRuns is sorted by created_at in descending order so the first element is the latest run
+	return &client.WorkflowRun{WorkflowId: *runs.WorkflowRuns[0].WorkflowID, RunId: *runs.WorkflowRuns[0].ID}, nil
+}
+
+func (gc *githubClient) GetWorkflowRunArtifacts(ctx context.Context, owner, repo, githubSBOMName string, runID int64) ([]*client.WorkflowArtifactContent, error) {
+	var res []*client.WorkflowArtifactContent
+
+	// get workflow run artifacts
+	artifacts, _, err := gc.ghClient.Actions.ListWorkflowRunArtifacts(ctx, owner, repo, runID, nil)
+	if err != nil {
+		return nil, fmt.Errorf("unable to list workflow run artifacts: %w", err)
+	}
+	for _, j := range artifacts.Artifacts {
+		// download artifact
```

Comment:
Check `(githubSBOMName != "" && *j.Name != githubSBOMName)` before download, and `continue` if so.

Reply:
Checked